docs/guides/go-built-in.md

```diff
@@ -311,112 +311,90 @@ information on why the transaction was rejected.
 
 Note that `CheckTx` does not execute the transaction, it only verifies that the transaction could be executed. We do not know yet if the rest of the network has agreed to accept this transaction into a block.
 
-
 Finally, make sure to add the bytes package to the `import` stanza at the top of `app.go`:
 
 ```go
 import(
-	"bytes"
+    "bytes"
 
-	"github.com/dgraph-io/badger/v3"
-	abcitypes "github.com/cometbft/cometbft/abci/types"
+    "github.com/dgraph-io/badger/v3"
+    abcitypes "github.com/cometbft/cometbft/abci/types"
 )
 ```
 
+### 1.3.3 FinalizeBlock
 
-### 1.3.3 BeginBlock -> DeliverTx -> EndBlock -> Commit
-
-When the CometBFT consensus engine has decided on the block, the block is transferred to the
-application over three ABCI method calls: `BeginBlock`, `DeliverTx`, and `EndBlock`.
+`FinalizeBlock` is an ABCI method introduced in CometBFT v0.38.0. This replaces the functionality provided previously (pre-v0.38.0) by the combination of ABCI methods `BeginBlock`, `DeliverTx`, and `EndBlock`.
```

Comment:
@lasarojc not sure if I understand your comment. The guide though introductory, is an "extension" of the previous 0.34 & 0.37 guide. 

Also, people will use 0.34 for some time before moving to 0.38. This might address their confusion about why `BeginBlock`, `DeliverTx`, and `EndBlock` are missing in 0.38. 

Or do you have a different concern?

Reply:
My concern is that this adds unnecessary complexity for someone just starting on CometBFT.
You are right that someone coming from the an older version would benefit from this comment, but in this case, it be better to have simpler "Upgrading to ABCI2.0" guide. 
But if we don't have bandwidth for that, then a compromise, in my opinion, would be to have these references to previous versions as notes as in

> :note:
> `FinalizeBlock` is an ABCI method introduced in CometBFT v0.38.0. This replaces the functionality provided previously (pre-v0.38.0) by the combination of ABCI methods `BeginBlock`, `DeliverTx`, and `EndBlock`.